flang/unittests/Optimizer/Builder/ComplexTest.cpp

```diff
@@ -0,0 +1,84 @@
+//===- ComplexExprTest.cpp -- ComplexExpr unit tests ----------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "flang/Optimizer/Builder/Complex.h"
+#include "gtest/gtest.h"
+#include "flang/Optimizer/Builder/FIRBuilder.h"
+#include "flang/Optimizer/Support/InitFIR.h"
+#include "flang/Optimizer/Support/KindMapping.h"
+
+struct ComplexTest : public testing::Test {
+public:
+  void SetUp() override {
+    fir::KindMapping kindMap(&context);
```

Comment:
Hi @kiranchandramohan , I just tested the PR in order to merge it and I am getting a crash in this test. I looked a little bit into it an it seems that `fir::KindMapping kindMap` should be a `std::unique_ptr` member of `ComplexTest` because the `firBuilder` ctor takes a reference to it, which is problematic given `kindMap` currently has the lifetime of the ctor.

Otherwise looks good and builds fine, thanks !